apps/sync/js/fxsync-webcrypto/fxsyncwebcrypto.js

```diff
@@ -0,0 +1,279 @@
+          obj.ciphertext);
+
+    try {
+      ret.IV =
+          StringConversion.base64StringToByteArray(obj.IV);
+    } catch (e) {
+      throw new Error('Could not parse IV as a base64 string');
+    }
+    try {
+      ret.hmacSignature =
+          StringConversion.hexStringToByteArray(obj.hmac);
+    } catch (e) {
+      throw new Error('Could not parse hmac as a hex string');
+    }
+    return ret;
+  }
+
+  /*
+   * setKeys - import kB and crypto/keys
+   *
+   * @param {String} kB Hex string with kB from FxA onepw protocol
+   * @param {Object} cryptoKeysStrings Object with:
+   *     - ciphertext {String} A Base64 String containing an AES-CBC ciphertext
+   *     - IV {String} A Base64 String containing the AES-CBC Initialization
+   *         Vector
+   *     - hmac {String} A Hex String containing the HMAC-SHA256 signature
+   * @returns {Promise} A promise that will resolve after import of kB and
+   *     decryption of cryptoKeys.
+   */
+  FxSyncWebCrypto.prototype.setKeys = function(kB, cryptoKeysStrings) {
+    var kBByteArray, cryptoKeys;
+
+    // Input checking
+    try {
+      kBByteArray = StringConversion.hexStringToByteArray(kB);
+    } catch (e) {
+      return Promise.reject('Could not parse kB as a hex string');
+    }
+
+    cryptoKeys = importFromStrings(cryptoKeysStrings);
+
+    return this._importKb(kBByteArray).then(function() {
+      return this._verifySyncKeys(cryptoKeys.hmacSignedText,
+          cryptoKeys.hmacSignature);
+    }.bind(this)).then(function(verified) {
+      if (verified) {
+        return this._importSyncKeys(cryptoKeys.IV,
+            cryptoKeys.ciphertext);
+      } else {
+        return Promise.reject(
+            'SyncKeys hmac could not be verified with current main key');
+      }
+    }.bind(this));
+  };
+
+  FxSyncWebCrypto.prototype.selectKeyBundle = function() {
+    return this.bulkKeyBundle.defaultAsKeyBundle;
+  };
+
+  /*
+   * decrypt - verify and decrypt a Weave Basic Object
+   *
+   * @param {Object} payloadStrings Object with:
+   *     - ciphertext {String} A Base64 String containing an AES-CBC ciphertext
+   *     - IV {String} A Base64 String containing the AES-CBC Initialization
+   *         Vector
+   *     - hmac {String} A Hex String containing the HMAC-SHA256 signature
+   * @param {String} collectionName String The name of the Sync collection
+   *     (currently ignored, see
+   *     https://github.com/michielbdejong/fxsync-webcrypto/issues/19)
+   * @returns {Promise} A promise for the decrypted Weave Basic Object.
+   */
+  FxSyncWebCrypto.prototype.decrypt = function(payloadStrings, collectionName) {
+    var keyBundle, payload;
+    if (typeof payloadStrings !== 'object') {
+      throw new Error('PayloadStrings is not an object');
+    }
+    if (typeof collectionName !== 'string') {
+      throw new Error('collectionName is not a string');
+    }
+    try {
+      keyBundle = this.selectKeyBundle(collectionName);
```

Comment:
You are not doing anything with `collectionName` on `selectKeyBundle`. Is this going to change when we add more collections? If that's not the case, maybe we can get rid of `selectKeyBundle`?